packages/angular_devkit/build_optimizer/src/transforms/class-fold.ts

```diff
@@ -61,8 +67,49 @@ export function getFoldFileTransformer(program: ts.Program): ts.TransformerFacto
             newBody,
           );
 
+          // Update remaining classes and statements.
+          statementsToRemove.push(...clazz.statements.map(st => st.expressionStatement));
+          classes = classes.filter(cl => cl != clazz);
+
           // Replace node with modified one.
-          return ts.visitEachChild(newNode, visitor, context);
+          return newNode;
+        }
+
+        // Check if node is a ES2015 class to replace with a pure IIFE.
+        clazz = classes.find((cl) => !cl.function && cl.declaration === node);
+        if (clazz) {
+          // There doesn't seem to be a way to make a statement from a class declaration.
+          // But it seems to work if you just jam it in there.
+          const classStatement = clazz.declaration as {} as ts.ExpressionStatement;
+          const innerReturn = ts.createReturn(ts.createIdentifier(clazz.name));
+
+          const iife = ts.createImmediatelyInvokedFunctionExpression([
+            classStatement,
+            ...clazz.statements.map(st => st.expressionStatement),
+            innerReturn,
+          ]);
+
+          const pureIife = ts.addSyntheticLeadingComment(
+            iife,
+            ts.SyntaxKind.MultiLineCommentTrivia,
+            '@__PURE__',
+            false,
+          );
+
+          // Move the original class modifiers to the var statement.
+          const newNode = ts.createVariableStatement(
+            clazz.declaration.modifiers,
+            ts.createVariableDeclarationList([
+              ts.createVariableDeclaration(clazz.name, undefined, pureIife),
+            ]),
```

Comment:
small nit: Maybe add `ts.NodeFlags.Const` as last arg to `ts.createVariableDeclarationList`

Reply:
Fixed